src/Future.js

```diff
@@ -10,6 +10,16 @@ function jail(handler, f){
   };
 }
 
+function fork(m, reject, resolve) {
+  if(m instanceof Future) {
+    return m._fork(reject, resolve);
+  } else if (typeof m.fork === 'function') {
+    return m.fork(reject, resolve);
+  } else {
+    reject(new TypeError('Expected a Future, got ' + R.toString(m)));
+  }
```

Comment:
It would have thrown "Undefined is not a function", may as well make it a little more clear while we're at it. :)